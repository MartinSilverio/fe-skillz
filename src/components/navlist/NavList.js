import React from 'react';
import NavItem from '../navitem/NavItem';

function NavList({ navItems, ...props }) {
  return (
    <ul {...props}>
      {navItems.map((navItem, i) => {
        return <NavItem key={navItem.id} navItem={navItem} />;
      })}
    </ul>
  );
}

export default NavList;
