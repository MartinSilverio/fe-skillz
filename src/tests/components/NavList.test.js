import React from 'react';
import { shallow } from 'enzyme';
import NavList from '../../components/navlist/NavList';

const itemList = [
  {
    title: 'Adipiscing',
    id: 31,
    selected: false,
    url: '/consectetur/adipiscing',
  },
  {
    title: 'Eiusmod',
    id: 32,
    selected: false,
    url: '/consectetur/eiusmod',
  },
  {
    title: 'Tempor',
    id: 33,
    selected: false,
    url: '/consectetur/tempor',
  },
];

const emptyList = [];

describe('NavList', () => {
  it('Should list NavItems if there are items available', () => {
    expect.assertions(1);
    expect(shallow(<NavList navItems={itemList} />)).toMatchSnapshot();
  });
  it('Should not list NavItems if there are not items available', () => {
    expect.assertions(1);
    expect(shallow(<NavList navItems={emptyList} />)).toMatchSnapshot();
  });
});
