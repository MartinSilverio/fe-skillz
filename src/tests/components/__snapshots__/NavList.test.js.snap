// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList Should list NavItems if there are items available 1`] = `
<ul>
  <NavItem
    key="31"
    navItem={
      Object {
        "id": 31,
        "selected": false,
        "title": "Adipiscing",
        "url": "/consectetur/adipiscing",
      }
    }
  />
  <NavItem
    key="32"
    navItem={
      Object {
        "id": 32,
        "selected": false,
        "title": "Eiusmod",
        "url": "/consectetur/eiusmod",
      }
    }
  />
  <NavItem
    key="33"
    navItem={
      Object {
        "id": 33,
        "selected": false,
        "title": "Tempor",
        "url": "/consectetur/tempor",
      }
    }
  />
</ul>
`;

exports[`NavList Should not list NavItems if there are not items available 1`] = `<ul />`;
